### PR TITLE
SYS-180 Remove logging from php(-fpm) - Redirect all output to journal

### PR DESCRIPTION
**Pull request by ppp0**

### CODE REVIEW

**Comment by ppp0** [comment]
This is quite big and it affects production. Needs careful testing - may be limited deployment to just one of the three nodes to see if all is good

File: modules/php5/templates/fpm/php-fpm.conf [hunk 1]
```diff
@@ -22,14 +22,14 @@
...
-pid = /var/run/php5-fpm.pid
+;pid = /var/run/php5-fpm.pid
 
 ; Error log file
 ; If it's set to "syslog", log is sent to syslogd instead of being written
 ; in a local file.
 ; Note: the default prefix is /var
 ; Default Value: log/php-fpm.log
-error_log = /var/log/php5-fpm/php5-fpm.log
+error_log = syslog
```
Comment (zazabe) [inline modules/php5/templates/fpm/php-fpm.conf]: as discussed, let's pass all php5-fpm config as argument for daemon in `init.pp` and keep all defaults here

File: modules/php5/templates/extension/opcache/conf.ini [hunk 2]
```diff
@@ -7,5 +7,5 @@ opcache.interned_strings_buffer=<%= @interned_strings_buffer %>
 opcache.max_accelerated_files=<%= @max_accelerated_files %>
 opcache.fast_shutdown=<%= @fast_shutdown ? 1 : 0 %>
 opcache.validate_timestamps=<%= @validate_timestamps ? 1 : 0 %>
-opcache.error_log=/var/log/php/opcache.log
+opcache.error_log=
```
Comment (zazabe) [inline modules/php5/templates/extension/opcache/conf.ini]: is it working?

File: modules/php5/templates/extension/newrelic/conf.ini [hunk 3]
```diff
@@ -1,8 +1,8 @@
 extension=newrelic.so
 newrelic.license = "<%= @license_key %>"
 newrelic.appname = "<%= @appname %>"
-newrelic.logfile = "/var/log/newrelic/php_agent.log"
-newrelic.daemon.logfile = "/var/log/newrelic/newrelic-daemon.log"
+newrelic.logfile = "/dev/stdout"
+newrelic.daemon.logfile = "/dev/stdout"
```
Comment (zazabe) [inline modules/php5/templates/extension/newrelic/conf.ini]: ok, but i guess those logs will have something like `journal.unit = php5.service`, could you check?

**Comment by zazabe** [comment]
We should have a test to check if all those logs are going to the right place

File: modules/php5/templates/extension/newrelic/conf.ini (hunk 3 above)
Comment (ppp0) [inline modules/php5/templates/extension/newrelic/conf.ini]: this is an excerpt of the journalclt:
```
Jun 20 08:10:37 debian-8-amd64-plain php5-gd[25148]: php5_invoke: Enable module gd for cli SAPI
Jun 20 08:10:46 debian-8-amd64-plain php5-imagick[26244]: php5_invoke: Enable module imagick for cli SAPI
Jun 20 08:10:51 debian-8-amd64-plain php5-curl[26793]: php5_invoke: Enable module curl for cli SAPI
Jun 20 08:10:53 debian-8-amd64-plain php5-mcrypt[26993]: php5_invoke: Enable module mcrypt for cli SAPI
Jun 20 08:10:54 debian-8-amd64-plain php5-xdebug[27078]: php5_invoke: Enable module xdebug for cli SAPI
[..]
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[13694]: php5_invoke: Enable module mongodb for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[13728]: php5_invoke: Enable module mysqli for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[13762]: php5_invoke: Enable module mongo for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[13796]: php5_invoke: Enable module pdo_mysql for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[13830]: php5_invoke: Enable module newrelic for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[13864]: php5_invoke: Enable module json for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[13898]: php5_invoke: Enable module memcache for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[13932]: php5_invoke: Enable module pdo for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[13966]: php5_invoke: Enable module readline for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14000]: php5_invoke: Enable module mcrypt for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14034]: php5_invoke: Enable module opcache for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14068]: php5_invoke: Enable module xdebug for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14102]: php5_invoke: Enable module gd for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14136]: php5_invoke: Enable module curl for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14170]: php5_invoke: Enable module intl for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14204]: php5_invoke: Enable module mysql for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14238]: php5_invoke: Enable module imagick for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14272]: php5_invoke: Enable module apcu for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14306]: php5_invoke: Enable module gearman for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14340]: php5_invoke: Enable module ssh2 for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14374]: php5_invoke: Enable module runkit for fpm SAPI
Jun 20 08:12:02 debian-8-amd64-plain php5-fpm[14408]: php5_invoke: Enable module stats for fpm SAPI
```

in JSON you actually see, that the unit is :( :
```json
"_SYSTEMD_UNIT" : "ssh.service"
```

**Comment by ppp0** [comment]
Grrr, NewRelic agent doesn't care about /dev/stdout or similar.. he just writes to his files in /var/log/new-relic/.. I'd say it's ok

**Comment by ppp0** [comment]
@{f4355e77-07da-4ca6-b033-d106f96ec3bb,Michel Meyer} shall I `in_tail` them? `/var/log/newrelic/newrelic-daemon.log` and `/var/log/newrelic/php_agent.log`? (log rotation is already in place, likely set up by the deb package)

**Comment by ppp0** [comment]
as discussed, let's add in_tail

**Comment by zazabe** [comment]
waiting for https://youtrack.cargomedia.ch/issue/SYS-184

File: modules/php5/templates/fpm/php-fpm.conf (hunk 1 above)
Comment (ppp0) [inline modules/php5/templates/fpm/php-fpm.conf]: Unfortunately, it is not possible to pass all these args via command line argg, cf. https://linux.die.net/man/8/php-fpm

**Comment by ppp0** [comment]
Testing was a pain here. Somehow, the JSON parser doesn't like the output from the box. I validated the results and did a test run with forwarding to Loggly.. There, all looks good
file

**Comment by ppp0** [comment]
retest this please

**Comment by zazabe** [comment]
should we be worried about php config changes? it will restart some prod services i guess...

**Comment by ppp0** [comment]
oh yes, we should deploy this scheduling a downtime. I will merge the additional extension timecop that has been added here in the meanwhile

File: modules/php5/templates/extension/opcache/conf.ini (hunk 2 above)
Comment (ppp0) [inline modules/php5/templates/extension/opcache/conf.ini]: see https://github.com/zendtech/ZendOptimizerPlus/blob/3b16ca512290b0416baf02962beedc703147a97a/README#L198
>Empty string assumes "stderr"

**Pull request merged**